generated-code-samples-for-review/language/language_classify_gcs.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by protoc-gen-go_gapic. DO NOT EDIT.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	language "cloud.google.com/go/language/apiv1"
+	languagepb "google.golang.org/genproto/googleapis/cloud/language/v1"
+)
+
+// [START language_classify_gcs]
+
+// sampleClassifyText: Classifying Content in text file stored in Cloud Storage
+//
+// gcsContentUri: Google Cloud Storage URI where the file content is located.
+// e.g. gs://[Your Bucket]/[Path to File]
+// The text file must include at least 20 words.
+func sampleClassifyText(gcsContentUri string) error {
+	ctx := context.Background()
+	c, err := language.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// gcsContentUri := "gs://cloud-samples-data/language/classify-entertainment.txt"
+	req := &languagepb.ClassifyTextRequest{
+		Document: &languagepb.Document{
+			Source: &languagepb.Document_GcsContentUri{
+				GcsContentUri: gcsContentUri,
+			},
+			// Available types: PLAIN_TEXT, HTML
+			Type: languagepb.Document_PLAIN_TEXT,
+			// Optional. If not specified, the language is automatically detected.
+			// For list of supported languages:
+			//   https://cloud.google.com/natural-language/docs/languages
+			Language: "en",
+		},
+	}
+	resp, err := c.ClassifyText(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	// Loop through classified categories returned from the API
+	for _, category := range resp.GetCategories() {
+		// Get the name of the category representing the document.
+		// See the predefined taxonomy of categories:
+		// https://cloud.google.com/natural-language/docs/categories
+		//
+		fmt.Printf("Category name: %v\n", category.GetName())
```

Comment:
Getters only really do anything on chained calls. I think we tend towards

`category.Name` rather than `category.GetName()`

Reply:
The decision to use getters was made by the previous author of the generator, to [properly deal with unset messages](https://github.com/googleapis/gapic-generator-go/blob/0023653f1848bde6f33e1e3f60e5ee95460572cb/internal/gensample/out.go#L377).

Reply:
sure, category is never nil here though.